inc/docker.h

```diff
@@ -34,6 +34,9 @@ char *docker_buffer(DOCKER *docker_client);
 CURLcode docker_delete(DOCKER *docker_client, char *url);
 CURLcode docker_post(DOCKER *docker_client, char *url, char *data);
 CURLcode docker_get(DOCKER *docker_client, char *url);
+CURLcode docker_delete_with_http_status(DOCKER *docker_client, char *url, long *http_status);
```

Comment:
Couple small things, I think the out-parameter will be non-obvious, so it's probably worth commenting about that (and probably providing documentating comments about more than just that eventually, but one thing at a time).

In particular I'd document that it is an out-parameter, and document the behavior if http_status is null.

Also consider naming it out_http_status so that it being intended as an out-param shows up in the declaration.